library.properties

```diff
@@ -6,5 +6,5 @@ sentence=Allows controlling the ST X-NUCLEO-IKS02A1 expansion board
 paragraph=This library provides several sample applications to control ST X-NUCLEO-IKS02A1 expansion board
 category=Sensors
 url=https://github.com/stm32duino/X-NUCLEO-IKS02A1
-architectures=stm32, avr, sam
-depends=STM32duino ISM330DHCX, STM32duino IIS2DLPC, STM32duino IIS2MDC
\ No newline at end of file
+architectures=stm32
```

Comment:
@cparata 
This means that all examples should not be used with other arch or simply that due to the new example it is not compatible? If only the new example is not compatible then simply add a guard like this on top of the sketch:
```C
#ifndef ARDUINO_ARCH_STM32
#error "This example is only for STM32 arch!"
#endif
```

Reply:
Hi @fpistm  ,
basically the audio part is strictly hardware dependent and at the moment it supports only F401RE and L476RG platforms, but in theory it can be easily extended to other platforms thanks to the way we structured the code. In fact the user needs only to implement a new pdm_low_level_xyz.cpp file for the new target platform where there are some weak functions that must be overridden. Because a default implementation (that returns an error code) of these weak functions is inside the PDM.cpp, the code should compile also with other platforms, not only stm32. Do you suggest to keep the support for other platforms like avr and samd on libraries.properties? Also because this limitation is present only for the audio part; the inertial examples should work also with other platforms.
Best Regards,
Carlo

Reply:
Thanks for explanation. So I guess only the sketch should warn user it is not compatible for other arch and also explain why to the user and he can implement the support for other arch.